src/lib/components/SpiderChart.svelte

```diff
@@ -0,0 +1,214 @@
+	}
+
+	function drawAnswerShape(answers: Record<string, number>): string | null {
+		let coordinates: [number, number][] = [];
+		for (var i = 0; i < features.length; i++) {
+			let ft_name = features[i];
+			let angle = Math.PI / 2 + (2 * Math.PI * i) / features.length;
+			let { x, y } = angleToCoordinate(angle, answers[ft_name]);
+			coordinates.push([x, y]);
+		}
+		return lineHelper(coordinates);
+	}
+
+	// draws circles for score labels
+	function getCircleCoords({ answer, idx }: { answer: number; idx: number }) {
+		let angle = Math.PI / 2 + (2 * Math.PI * idx) / features.length;
+		return {
+			...angleToCoordinate(angle, answer),
+			answer
+		};
+	}
+	const formattedAnswers = $derived(
+		features.reduce(
+			(
+				resultArray: {
+					feature: string;
+					answer: number;
+					xCoord: number;
+					yCoord: number;
+					idx: number;
+				}[],
+				feature,
+				idx
+			) => {
+				const answer = answers[feature];
+				const coords = getCircleCoords({ answer, idx });
+				resultArray.push({ feature, answer, xCoord: coords.x, yCoord: coords.y, idx });
+				return resultArray;
+			},
+			[]
+		)
+	);
+
+	// calculates the offset of each label's centerpoint from the center of the chart
+	let labels = $derived.by(() => {
+		const numPoints = features.length;
+		const points = [];
+		const angleStep = (2 * Math.PI) / numPoints;
+		// not using radialScale because we want a radius outside those bounds
+		const radius = config.d / 2 + config.p + config.labelD / 2;
+
+		for (let i = 0; i < numPoints; i++) {
+			const angle = i * angleStep;
+			const x = radius * Math.sin(angle);
+			// multiply y by -1 so it maps clockwise
+			const y = radius * Math.cos(angle) * -1;
+			points.push({ offsetX: x, offsetY: y, text: dynamics[i] });
+		}
+		return points;
+	});
+</script>
+
+<svelte:window bind:innerWidth />
+<div class="outer" style={`padding: ${config.labelD}px;`}>
+	<svg id="chart" width={config.d + 2 * config.p} height={config.d + 2 * config.p}>
+		<g id="ticks">
+			{#each config.ticks as tick}
+				<!-- concentric octogons -->
+				<polygon points={tickToPolygon(tick)} />
+			{/each}
+			{#each radialTickLines() as f}
+				<line
+					x1={config.p + config.d / 2}
+					y1={config.p + config.d / 2}
+					x2={f.outerX}
+					y2={f.outerY}
+				/>
+			{/each}
+		</g>
+		<g id="answer">
+			<path stroke-width="3" opacity="0.8" d={drawAnswerShape(answers)} />
+			{#each formattedAnswers as ans}
+				<circle cx={ans.xCoord} cy={ans.yCoord} r={config.p}></circle>
+				<text x={ans.xCoord - 5} y={ans.yCoord + 3}>{ans.answer}</text>
+			{/each}
+		</g>
+	</svg>
+	<div class="labels-container" style={`width: ${config.labelD}px; height: ${config.labelD}px;`}>
+		{#each labels as label, idx}
+			<div
+				class="dynamic"
+				style={`width: ${config.labelD}px; height: ${config.labelD}px; transform: translate(${label.offsetX}px, ${label.offsetY}px)`}
+			>
+				{#if innerWidth >= BREAKPOINT}
+					{label.text}
+				{:else}
+					Dynamic {idx + 1}
```

Comment:
Love having this shortened string for small screens! Not for this PR, but I wonder if a short keyword string would work for each dynamic, or maybe a popup on click to see the full name.